vendor/runner/runner.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const glob = require('glob');
+const uniq = require('lodash.uniq');
+
+/**
+ * Load a collection of ServiceTester objects and register them with Mocha.
+ */
+class Runner {
+  /**
+   * Function to invoke before each test. This is a stub which can be
+   * overridden on instances.
+   */
+  beforeEach () {}
+
+  /**
+   * Prepare the runner by loading up all the ServiceTester objects.
+   */
+  prepare () {
+    this.testers = glob.sync(`${__dirname}/../*.js`).map(require);
+    this.testers.forEach(tester => {
+      tester.beforeEach = () => { this.beforeEach(); };
+    });
+  }
+
+  _testersWithVendor (vendor) {
+    return this.testers.filter(t => t.id.toLowerCase() === vendor);
+  }
+
+  /**
+   * Limit the test run to the specified vendors.
+   *
+   * @param vendors An array of vendor names to run
+   */
+  only (vendors) {
+    const normalizedVendors = uniq(vendors.map(v => v.toLowerCase()));
```

Comment:
Neat!

I don't like to vendor in that kind of stuff because I've seen it attract modifications. 😝 

Fortunately lodash ships packages for the individual functions.